feat: header & search page 구현

euijinkk opened the pull request:
### Summary 
- header bar 구현
- search page 구현
- header 속 search 버튼에 search page link -> Router안에 header 넣음.

euijinkk committed:
feat: header & search page 구현

juno7803 commented on src/components/index.js:
```diff
@@ -1,4 +1,4 @@
 export { default as ScrollSlider } from './main/ScrollSlider';
-export { default as Writers } from './main/Writers';
\ No newline at end of file
+export { default as Writers } from './main/Writers';
```
`writers.js`가 아직 `Writers.js`로 안바뀐거같아! 수정 부탁할게 :)
컴포넌트 이름은 대문자로 시작!! (Pascal Case)

euijinkk replied:
바꾸었는데 왜 안바뀐채로 넘어가지 .. ㅠ 다시 바꿨습니다!

juno7803 commented on src/pages/SearchPage.js:
```diff
@@ -1,14 +1,120 @@
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  height: 100vh;
+
+  .search {
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    /* 왜 padding을 높게 줘야, 화면이 상단에 고정될까?  */
+    padding: 30rem 3rem;
+    position: fixed;
```
음 지금 해봤는데 패딩은 왜 있는지 나두 잘 모르겠어 ㅠㅠ
```css
height: 100%;

.search {
  position: fixed;
}
```
만 해도 스크롤 안생기구 가운데 오는 것 같아! (`padding` 빼고! 주석도 지워주세요 :))

euijinkk replied:
아 100%는 추가했습니다. vh 하니까 스크롤이 생기네요. 
padding을 넣은 이유는! 화면이 줄어들었을때에도 언제나 header가 맨 위에 있고, 그 밑에 search bar 가 있게 하려고 한거에요! padding을 높게 줘야 그게 되더라구용..? 실제로 brunch에 그렇게 되어 있어서 따라해봤습니다.

juno7803 replied:
오키오키! 이건 한번 더 확인해볼게요 👍

juno7803 commented on src/components/main/writers.js:
```diff
@@ -1,4 +1,9 @@
+/* eslint-disable react/prop-types */
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable react/button-has-type */
```
음 나는 이거 지우고 돌려도 의진이가 `eslintrc.json` 에 쓴것만으로도 잘 돌아가더라구! 그래서 한번더 yarn start 해보고 그래도 안돌아가면 남기고, 아니라면 지우자!

juno7803 commented on .eslintrc.json:
```diff
@@ -11,7 +11,10 @@
     "react/jsx-filename-extension": ["off"],
     "react/jsx-one-expression-per-line": ["off"],
-    "react/react-in-jsx-scope": ["off"]
+    "react/react-in-jsx-scope": ["off"],
+    // 밑에 두개 추가해도, 적용이 안되네요?.. 맨위에 주석 띄우라 해서 그렇게 뒀습니다.
+    "react/button-has-type": ["off"],
+    "no-unused-vars": ["off"]
```
나는 밑에 `/**/` 로 컴포넌트마다 남긴 부분 지우더라도 잘 되던데 왜 그런지 모르겠다 ㅠㅠ

euijinkk replied:
일단 두고,, 나중에 다시 되면 지울게요!

juno7803 replied:
좋아요 :)

juno7803 commented on src/components/common/Header.js:
```diff
@@ -1,7 +1,62 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable react/prop-types */
 import React from 'react';
+import { withRouter } from 'react-router-dom';
```
**<< 참고 >>**
세미나땐 `withRouter`를 배우긴 했지만, 요즘 HOC(고차 컴포넌트 - 컴포넌트를 다른 컴포넌트로 감싸는 것) 를 잘 쓰는 추세는 아니야!
그래서 난 사실 `react-router-dom` 의 `useHistory`를 사용해!
```jsx
import { useHistory } from 'react-router-dom';

const Header = () => {
  const history = useHistory();
}
// 이후엔 똑같이 사용
```
정말 쉽지?? react-router-dom 안에 내장된 훅인데다가 굳이 HOC 패턴을 쓰지 않아도 되니까 난 요걸 애용하는편! 
꼭 바꿀 필요는 없고 이런것도 있다고만 알아두기!

https://ko.reactjs.org/docs/higher-order-components.html

euijinkk replied:
미쳐따... 대박 편하네요!! 바꿔서 적용해보았어요~

juno7803 commented on src/pages/SearchPage.js:
```diff
@@ -1,14 +1,120 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable react/button-has-type */
```
얘도 한번 더 확인해보고 되면 지우고 아니면 남기기

euijinkk replied:
왠지 모르겠는데 안돼서, 일단 남겨둘게요!

juno7803 reviewed (changes requested):
`Footer.js` 에서
```jsx
const Footer = () => {
  if (window.location.pathname === '/search') return null;
  return ();
}
```
이렇게 바꾸면, `'/search'` 에선 Footer가 보이지 않을거야! 이렇게 수정해서 다시 푸시해줘!

euijinkk committed:
feat: pr 리뷰 반영

1. search page 에서 footer 보이지 않도록 구현
2. search page에서 스크롤 없앰.
3. withRouter을 useHistory로 수정

euijinkk committed:
feat: 주석제거, eslintrc 수정
